test/metabase/metabot/metabot_util_test.clj

```diff
@@ -155,26 +157,52 @@
-        (is (str/includes? create_database_ddl "create type PEOPLE_SOURCE_t as enum 'Affiliate', 'Facebook', 'Google', 'Organic', 'Twitter';"))
-        (is (str/includes? create_database_ddl "create type REVIEWS_RATING_t as enum '1', '2', '3', '4', '5';"))
-        (is (str/includes? create_database_ddl "CREATE TABLE \"PRODUCTS\" ("))
-        (is (str/includes? create_database_ddl "CREATE TABLE \"ORDERS\" ("))
-        (is (str/includes? create_database_ddl "CREATE TABLE \"PEOPLE\" ("))
-        (is (str/includes? create_database_ddl "CREATE TABLE \"REVIEWS\" ("))))))
+      (tu/with-temporary-setting-values [metabot-settings/enum-cardinality-threshold 50]
+        (let [{:keys [create_database_ddl]} (metabot-util/denormalize-database {:id (mt/id)})]
+          (is (str/includes? create_database_ddl "create type PRODUCTS_CATEGORY_t as enum 'Doohickey', 'Gadget', 'Gizmo', 'Widget';"))
+          (is (str/includes? create_database_ddl "create type PEOPLE_STATE_t as enum 'AK', 'AL', 'AR', 'AZ', 'CA', 'CO', 'CT',"))
+          (is (str/includes? create_database_ddl "create type PEOPLE_SOURCE_t as enum 'Affiliate', 'Facebook', 'Google', 'Organic', 'Twitter';"))
+          (is (str/includes? create_database_ddl "create type REVIEWS_RATING_t as enum '1', '2', '3', '4', '5';"))
+          (is (str/includes? create_database_ddl "CREATE TABLE \"PRODUCTS\" ("))
+          (is (str/includes? create_database_ddl "CREATE TABLE \"ORDERS\" ("))
+          (is (str/includes? create_database_ddl "CREATE TABLE \"PEOPLE\" ("))
+          (is (str/includes? create_database_ddl "CREATE TABLE \"REVIEWS\" (")))))))
+
+(deftest inner-query-test
+  (testing "Ensure that a dataset-based query contains AS aliases"
+    (mt/dataset sample-dataset
+      (t2.with-temp/with-temp
+        [Card orders-model {:name    "Orders Model"
+                            :dataset_query
+                            {:database (mt/id)
+                             :type     :query
+                             :query    {:source-table (mt/id :orders)}}
+                            :dataset true}]
+        (let [q (metabot-util/inner-query orders-model)
+              [_ aliases] (re-matches #"(?ism)SELECT(.+)FROM(:?.*)" q)]
+          (is (true? (str/includes? aliases " AS ")))))))
+  (testing "Ensure that a native dataset query contains a SELECT *"
+    (mt/dataset sample-dataset
+      (t2.with-temp/with-temp
+        [Card orders-model {:name          "Orders Model"
+                            :dataset_query {:database (mt/id)
+                                            :type     :native
+                                            :native   {:query "SELECT COUNT(*) FROM ORDERS;"}}
+                            :dataset       true}]
+        (let [q (metabot-util/inner-query orders-model)
+              [_ aliases] (re-matches #"(?ism)SELECT(.+)FROM(:?.*)" q)]
+          (is (= "*" (str/trim aliases))))))))
```

Comment:
I'd like to see some more tests around what happens with models with joins and maybe ones with aggregations. eg,

```clojure
      (t2.with-temp/with-temp
        [Card joined-model {:dataset true
                            :database_id (mt/id)
                            :query_type :query
                            :dataset_query
                            (mt/mbql-query orders
                                           {:fields [$total &products.products.category]
                                            :joins [{:source-table $$products
                                                     :condition [:= $product_id &products.products.id]
                                                     :strategy :left-join
                                                     :alias "products"}]})}]
        (is (= "???" (#'metabot-util/aliases joined-model))))
```

This also raises an idea of where the string joining should be. Not sure if you want the `aliases` function to return a vector of `[col-name display-name]` to check here or something similar.